webrob/neems/manager.py

```diff
@@ -1,23 +1,39 @@
 import os
 from flask import session
 from webrob.neems.neem import NEEM
-from webrob.app_and_db import app, mongoDBMetaCollection
+from webrob.app_and_db import app, get_mongo_db_meta_collection
+from pymongo.errors import ConnectionFailure, PyMongoError
+from webrob.models.NEEMMetaException import NEEMMetaException
+from webrob.models.NEEMHubSettings import get_settings_count
 
 NEEM_DIR = "/neems"
 
 
 class NEEM_Manager:
     def __init__(self):
-        self.neem_ids = self.__get_neem_ids__()
-        # TODO: remove again, just for testing pagination as long as only few neems
-        #        are there
-        self.neem_ids = self.neem_ids * 20
+        self.neem_ids = self.set_neem_ids()
+
+    def set_neem_ids(self):
+        self.neem_ids = []
 
-    def __get_neem_ids__(self):
-        neem_ids = []
         # get all neem ids information from collection
-        neem_ids = mongoDBMetaCollection.find().distinct('_id')
-        return neem_ids
+        # first find if there is any existing entry in db
+        count = get_settings_count()
+        if count > 0:
+            try:
+                mongoDBMetaCollection = get_mongo_db_meta_collection()
+                self.neem_ids = mongoDBMetaCollection.find().distinct('_id')
+                # TODO: remove again multiply with 20 and return only self.neem_ids,
+                #  just for testing pagination as long as only few neems are there
+                self.neem_ids = self.neem_ids * 20
+
+            except ConnectionFailure as e:
```

Comment:
why not adding a flask handler for this error?

Reply:
done. Now ConnectionFailure has its own flask handler.